src/map/achievement.cpp

```diff
@@ -510,21 +498,17 @@ int *achievement_level(struct map_session_data *sd, bool flag)
 
 /**
  * Update achievement objectives.
- * @see DBApply
+ * @param sd: Player to update
+ * @param ad: Achievement data to compare for completion
+ * @param group: Achievement group to update
+ * @param update_count: Objective values player has
+ * @return 1 on success and false on failure
  */
-static int achievement_update_objectives(DBKey key, DBData *data, va_list ap)
+static int achievement_update_objectives(struct map_session_data *sd, struct s_achievement_db *ad, enum e_achievement_group group, int update_count[MAX_ACHIEVEMENT_OBJECTIVES])
```

Comment:
Transfert that array via it's adress / ref instead ? (Could be wrap with std::array if we want to keep a fixed size)